styx-client/src/main/java/com/spotify/styx/client/auth/GoogleIdTokenAuth.java

```diff
@@ -70,9 +88,24 @@ private String getToken(String targetAudience, GoogleCredential credential)
     }
   }
 
+  private String getDefaultGCEIdToken(String targetAudience) throws IOException {
+    // https://cloud.google.com/compute/docs/instances/verifying-instance-identity#request_signature
+    final String metadataHost = System.getenv().getOrDefault("GCE_METADATA_HOST", DEFAULT_GCE_METADATA_HOST);
+    final String uriTemplate = "http://" + metadataHost + GCE_METADATA_IDENTITY_PATH;
```

Comment:
Given the default value
```java
private static final String DEFAULT_GCE_METADATA_HOST = "http://169.254.169.254";
```

this will prefix `http://` twice, leading to url like `http://http://169.254.169.254`